docs/solidity-by-example.rst

```diff
@@ -620,4 +620,52 @@ Safe Remote Purchase
 Micropayment Channel
 ********************
 
-To be written.
+::
+
+    pragma solidity ^0.4.11;
+    contract Channel {
+        address[2] users;
+        uint deposit;
+        uint timeout;
+        uint closeTimestamp;
+        uint closeSequenceNumber;
+        uint closeBalanceUser0;
+        enum State { Opened, Accepted, Closing, Closed }
+        State state;
+        function Channel(address _other, uint _timeout) payable {
+            users[0] = msg.sender;
+            users[1] = _other;
+            deposit = msg.value;
+            require(deposit < 2**250);
+            require(_timeout < 2 days);
+            timeout = _timeout;
+        }
+        function accept() payable {
+            require(msg.sender == users[1]);
```

Comment:
Since `user[0]` and `user[1]` have a very specific role, I think it would make sense naming them appropriately.